aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1411,6 +1411,10 @@ Tensor& mean_out(const Tensor& self, DimnameList dim,
   return at::mean_out(result, self, dimnames_to_positions(self, dim), keepdim, opt_dtype);
 }
 
+Tensor& mean_dtype_out(const Tensor &self, std::optional<ScalarType> dtype, Tensor& result) {
+  return at::mean_out(result, self, IntArrayRef{}, false, dtype);
```

Comment:
Looking at the CI failure here: https://github.com/pytorch/pytorch/actions/runs/10410262594/job/28833890475?pr=133506

I tried running a similar OpInfo test that passes (`python test/test_ops.py TestCommonCUDA.test_out_stack_cuda_float32`) and seeing how that exception is properly being raised.

For `stack()`, it looks like it comes from this check in the meta function: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/TensorShape.cpp#L284

So you can try adding a similar assert in your kernel here?